src/components/common/ButtonWithIcon.vue

```diff
@@ -1,27 +1,42 @@
 <template>
   <BaseButton
+    class="flex items-center"
     :type="type"
     :disabled="disabled"
     :size="size"
   >
-    <div class="flex items-center">
-      <slot name="icon" />
-      <slot name="text" />
-    </div>
+    <fa
+      :class="classes"
+      class="mr-3"
```

Comment:
I find this margin a bit too much. See screenshot
![image](https://user-images.githubusercontent.com/30908158/132967535-d0fd6dc2-e87e-4618-b2d9-80321d5927b6.png)

However I think this is a bit more subjective.